components/kyma-environment-broker/internal/storage/postsql/read.go

```diff
@@ -541,38 +558,95 @@ func (r readSession) getInstanceCount(filter dbmodel.InstanceFilter) (int, error
 	var res struct {
 		Total int
 	}
-	stmt := r.session.Select("count(*) as total").From(InstancesTableName)
+	var stmt *dbr.SelectStmt
+	if len(filter.States) == 0 {
+		stmt = r.session.Select("count(*) as total").From(InstancesTableName)
+	} else {
+		stmt = r.session.
+			Select("count(*) as total").
+			From(InstancesTableName).
+			Join(dbr.I(OperationTableName).As("o1"), fmt.Sprintf("%s.instance_id = o1.instance_id", InstancesTableName)).
+			LeftJoin(dbr.I(OperationTableName).As("o2"), fmt.Sprintf("%s.instance_id = o2.instance_id AND o1.created_at < o2.created_at AND o2.state <> '%s'", InstancesTableName, orchestration.Pending)).
+			Where("o2.created_at IS NULL").
+			Where(fmt.Sprintf("o1.state <> '%s'", orchestration.Pending))
+
+		stateFilters := buildInstanceStateFilters("o1", filter)
+		stmt.Where(stateFilters)
+	}
+
 	addInstanceFilters(stmt, filter)
 	err := stmt.LoadOne(&res)
 
 	return res.Total, err
 }
 
+func buildInstanceStateFilters(table string, filter dbmodel.InstanceFilter) dbr.Builder {
+	exprs := []dbr.Builder{}
+	for _, f := range filter.States {
+		switch f {
```

Comment:
nit: we're iterating over `States`, consider changing variable name to `s` then

```suggestion
	for _, s := range filter.States {
		switch s {
```